src/Gui/MDIView.cpp

```diff
@@ -377,81 +401,116 @@ void MDIView::changeEvent(QEvent *e)
     }
 }
 
+bool MDIView::eventFilter(QObject* watched, QEvent* e)
+{
+    // As long as this widget is a top-level window (either in 'TopLevel' or 'FullScreen' mode) we
+    // need to be notified when an action is added to a widget. This action must also be added to
+    // this window to allow one to make use of its shortcut (if defined).
+    // Note: We don't need to care about removing an action if its parent widget gets destroyed.
+    // This does the action itself for us.
+
+    if (watched != this && e->type() == QEvent::ActionAdded) {
+        auto a = static_cast<QActionEvent*>(e);
+        QAction* action = a->action();
+
+        if (!action->isSeparator()) {
+            QList<QAction*> acts = actions();
+            if (!acts.contains(action)) {
+                addAction(action);
+            }
+        }
+    }
+
+    return false;
+}

```

Comment:
Let's avoid one letter variable names `event` and `actionEvent` would be better suited.